osu.Game/Screens/Play/SkipOverlay.cs

```diff
@@ -143,24 +143,35 @@ protected override void LoadComplete()
         /// </summary>
         public void SkipWhenReady()
         {
-            if (skipQueued) return;
+            if (SkipQueued) return;
 
-            skipQueued = true;
+            SkipQueued = true;
             attemptNextSkip();
 
             void attemptNextSkip() => Scheduler.AddDelayed(() =>
             {
-                if (!button.Enabled.Value)
+                if (!isClickable || !gameplayClock.IsRunning)
                 {
-                    skipQueued = false;
+                    attemptNextSkip();
                     return;
                 }
 
                 button.TriggerClick();
-                attemptNextSkip();
+
+                double progress = getProgress();
+
+                if (progress > 0)
+                    attemptNextSkip();
+                else
+                    SkipQueued = false;
             }, 200);
         }
 
+        private double getProgress()
+        {
+            return Math.Max(0, 1 - (gameplayClock.CurrentTime - displayTime) / (fadeOutBeginTime - displayTime));
+        }
```

Comment:
I would just make this a normal expression-bodied property at this point.

```csharp
private double progress => Math.Max(0, 1 - (gameplayClock.CurrentTime - displayTime) / (fadeOutBeginTime - displayTime));
```

(not a suggestion because usages would need updating too)